bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java

```diff
@@ -321,30 +322,10 @@ private void handleMultiChannelCapabilityReportResponse(SerialMessage serialMess
 			}
 
 			// Add the device classes
-			Basic basic = this.getNode().getDeviceClass().getBasicDeviceClass();
-			Generic generic = Generic.getGeneric(genericDeviceClass);
-			if (generic == null) {
-				logger.error(String.format("NODE %d: Endpoint %d has invalid device class. generic = 0x%02x, specific = 0x%02x.", 
-						this.getNode().getNodeId(), endpoint, genericDeviceClass, specificDeviceClass));
+			if (!updateDeviceClass(endpoint, genericDeviceClass, specificDeviceClass, dynamic)) {
+				// Updating device class failed, already logged, continue with next endpoint
 				continue;
 			}
-			Specific specific = Specific.getSpecific(generic, specificDeviceClass);
```

Comment:
The code is moved (extracted). To improve readability. No functional change in this commit.